include/engine/engine_config.hpp

```diff
@@ -84,6 +84,7 @@ struct EngineConfig final
     int max_locations_viaroute = -1;
     int max_locations_distance_table = -1;
     int max_locations_map_matching = -1;
+    boost::optional<double> max_radius_map_matching = -1;
```

Comment:
Should this be a plain double similarly to how other limits above are plain ints? You could initialize with `std::numeric_limits<double>::max()` if you're concerned about the unlimited case.